src/main/java/com/commercetools/sync/types/utils/FieldDefinitionUpdateActionUtils.java

```diff
@@ -0,0 +1,110 @@
+            .of(
+                buildChangeLabelUpdateAction(oldFieldDefinition, newFieldDefinition)
+            )
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(toList());
+
+
+        if (isPlainEnumField(oldFieldDefinition)) {
+            updateActions.addAll(buildEnumValuesUpdateActions(
+                    oldFieldDefinition.getName(),
+                    ((EnumFieldType) oldFieldDefinition.getType()).getValues(),
+                    ((EnumFieldType) newFieldDefinition.getType()).getValues()
+            ));
+        } else if (isLocalizedEnumField(oldFieldDefinition)) {
+            updateActions.addAll(buildLocalizedEnumValuesUpdateActions(
+                    oldFieldDefinition.getName(),
+                    ((LocalizedEnumFieldType) oldFieldDefinition.getType()).getValues(),
+                    ((LocalizedEnumFieldType) newFieldDefinition.getType()).getValues()
+            ));
+        }
+
+
+        return updateActions;
+    }
+
+    /**
+     * Indicates if the field is a plain enum value or not.
+     *
+     * @param fieldDefinition the field definition.
+     * @return true if the field definition is a plain enum value, false otherwise.
+     */
+    private static boolean isPlainEnumField(@Nonnull final FieldDefinition fieldDefinition) {
+        return fieldDefinition.getType().getClass() == EnumFieldType.class;
+    }
+
+    /**
+     * Indicates if the field definition is a localized enum value or not.
+     *
+     * @param fieldDefinition the field definition.
+     * @return true if the field definition is a localized enum value, false otherwise.
+     */
+    private static boolean isLocalizedEnumField(@Nonnull final FieldDefinition fieldDefinition) {
+        return fieldDefinition.getType().getClass() == LocalizedEnumFieldType.class;
+    }
+
+    /**
+     * Compares the {@link LocalizedString} labels of an {@link FieldDefinition} and an
+     * {@link FieldDefinition} and returns an {@link UpdateAction}&lt;{@link Type}&gt; as a result in
+     * an {@link Optional}. If both the {@link FieldDefinition} and the {@link FieldDefinition} have the
+     * same label, then no update action is needed and hence an empty {@link Optional} is returned.
+     *
+     * @param oldFieldDefinition the field definition which should be updated.
+     * @param newFieldDefinition the field definition draft where we get the new label.
+     * @return A filled optional with the update action or an empty optional if the labels are identical.
+     */
+    @Nonnull
+    public static Optional<UpdateAction<Type>> buildChangeLabelUpdateAction(
+        @Nonnull final FieldDefinition oldFieldDefinition,
+        @Nonnull final FieldDefinition newFieldDefinition) {
+
+        return buildUpdateAction(oldFieldDefinition.getLabel(), newFieldDefinition.getLabel(),
+            () -> ChangeFieldDefinitionLabel.of(oldFieldDefinition.getName(), newFieldDefinition.getLabel())
```

Comment:
I think is must be oldFieldDefinition.getLabel()

Reply:
yes u are right about the confusion 👍 but the first parameter is fieldName.